inference-engine/include/ie_plugin_config.hpp

```diff
@@ -143,6 +143,16 @@ DECLARE_METRIC_KEY(NUMBER_OF_WAITING_INFER_REQUESTS, unsigned int);
  */
 DECLARE_METRIC_KEY(NUMBER_OF_EXEC_INFER_REQUESTS, unsigned int);
 
+/**
+ * @brief Metric which defines the device architecture.
+ */
+DECLARE_METRIC_KEY(DEVICE_ARCHITECTURE, std::string);
```

Comment:
CPU, GPU are devices names, while `DEVICE_ARCHITECTURE` differentiates different devices of the same family (e.g. MYX or MY2, while device name is still MYRIAD)

Reply:
We have `FULL_DEVICE_NAME`, which can include this information: ` Intel (r) Neural Compute Stick 2` ~ `MYX`. Also device name itself can include this infromation: `MYRIAD.ma2480-0` ~ `MYX` on USB port 0.

Reply:
In this case core part shall be able to parse <id> part which is plugin-specific. So, on core side it is not possible to understand that "ma2480-0" and "ma2480-1" belong to same architecture. So, it is better on plugin side to parse this "DEVICE_ID" and return "ma2480" for both devices and it will indicate that same cache can be reused

Reply:
@vinograd47 for some devices `FULL_DEVICE_NAME` provides marketing name and we don't know which architecture does it have. E. g. CPUs or GPU with different `FULL_DEVICE_NAME` may consume the same binaries.